test/stability.js

```diff
@@ -88,3 +89,52 @@ describe("TestRPC", function(done) {
     batch.execute();
   });
 });
+
+describe("race conditions", function(done) {
+  var web3 = new Web3();
+  var provider;
+  var accounts;
+
+  before("initialize the provider", function() {
+    provider = TestRPC.provider();
+    web3.setProvider(provider);
+  });
+
+  before("get accounts", function(done) {
+    web3.eth.getAccounts(function(err, accs) {
+      if (err) return done(err);
+
+      accounts = accs;
+      done();
+    });
+  });
+
+  it("should not cause 'get' of undefined", function(done) {
+    process.prependOnceListener('uncaughtException', function(err) {
```

Comment:
These tests demonstrate the problem beautifully, but if the problem is fixed, they'll hang... Ideally this would use `assert.throws`, but I can imagine that not being possible if the error is thrown on a function run by `setImmediate` or similar (my big gripe with the `async` module).

At the very least, perhaps we can call `done()` on a timeout? That feels a bit hokey, but if there really is no way to directly catch the exception, I think that's all that can be done, right?

Reply:
Actually, I think that's a fine way to do it (with `setTimeout`, because even in the cases where the "it's fixed" timeout fires before the uncaught exception handler is thrown, mocha will complain to us that `done` was called twice, which in this case indicates a (late) test failure.

Reply:
Doh, forget I said that... You can just call `done()` from the `revert` callback on line 120, and the `call` callback on line 135.

Reply:
I'll just make this change on merge.